### PR TITLE
Update CHANGELOG and release notes

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Prepares the release notes and changelog for 2.8.1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,19 @@
...
 * [9185](https://github.com/grafana/loki/pull/9185) **dannykopping**: Prevent redis client from incorrectly choosing cluster mode with local address.
+* [9176](https://github.com/grafana/loki/pull/9176) **DylanGuedes**: Fix incorrect association of per-stream rate limit when sharding is enabled.
+
+##### Changes
+
+* [9106](https://github.com/grafana/loki/pull/9106) **trevorwhitney**: Update go to 1.20.3
+
+#### Promtail
+
+##### Fixes
```
Comment (JordanRushing) [inline CHANGELOG.md:17]:
```suggestion
##### Fixes

```

We are adding whitespace here in other sections

File: docs/sources/release-notes/v2-8.md [hunk 2]
```diff
@@ -21,3 +21,12 @@ For a full list of all changes please look at the [CHANGELOG](https://github.com
 As always, please read the [upgrade guide]({{<relref "../upgrading/#270">}}) before upgrading Loki.
 
 ## Bug fixes
+
+### 2.8.1 (2023-04-21)
+
+
+* Fix bug that was dropping index if period is a zero value
```
Comment (JordanRushing) [inline docs/sources/release-notes/v2-8.md:28]:
```suggestion
* Fix bug that was dropping index if period is a zero value.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,8 +3,20 @@
 #### Loki
 
 ##### Fixes
+
 * [9156](https://github.com/grafana/loki/pull/9156) **ashwanthgoli**: Expiration: do not drop index if period is a zero value
```
Comment (JordanRushing) [inline CHANGELOG.md:7]:
```suggestion
* [9156](https://github.com/grafana/loki/pull/9156) **ashwanthgoli**: Expiration: do not drop index if period is a zero value.
```

**Review by JordanRushing — APPROVED** [review]

File: docs/sources/release-notes/v2-8.md [hunk 4]
```diff
@@ -21,3 +21,12 @@ For a full list of all changes please look at the [CHANGELOG](https://github.com
...
 
 ## Bug fixes
+
+### 2.8.1 (2023-04-21)
+
+
+* Fix bug that was dropping index if period is a zero value.
+* Fix redis client to prevent it from incorrectly choosing cluster mode with local address.
+* Fix incorrect association of per-stream rate limit when sharding is enabled.
+* Update go to 1.20.3 to address security vulnerabilities if previous go version.
```
Comment (vlad-diachenko) [inline docs/sources/release-notes/v2-8.md:31]:
```suggestion
* Update go to 1.20.3 to address security vulnerabilities in the previous go version.
```

**Review by vlad-diachenko — APPROVED** [review]
lgtm

**Comment by vlad-diachenko** [comment]
also, we need to merge it to main and backport it back to release-2.8.x branch. or something like this to have this changes in `main` also

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,8 +3,20 @@
...
 ##### Fixes
-* [9156](https://github.com/grafana/loki/pull/9156) **ashwanthgoli**: Expiration: do not drop index if period is a zero value
+
+* [9156](https://github.com/grafana/loki/pull/9156) **ashwanthgoli**: Expiration: do not drop index if period is a zero value.
 * [9185](https://github.com/grafana/loki/pull/9185) **dannykopping**: Prevent redis client from incorrectly choosing cluster mode with local address.
+* [9176](https://github.com/grafana/loki/pull/9176) **DylanGuedes**: Fix incorrect association of per-stream rate limit when sharding is enabled.
+
+##### Changes
+
+* [9106](https://github.com/grafana/loki/pull/9106) **trevorwhitney**: Update go to 1.20.3
```
Comment (JStickler) [inline CHANGELOG.md:13]:
```suggestion
* [9106](https://github.com/grafana/loki/pull/9106) **trevorwhitney**: Update go to 1.20.3.
```

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Pull request merged**